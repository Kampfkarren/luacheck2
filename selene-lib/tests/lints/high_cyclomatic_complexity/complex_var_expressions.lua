local x = {}
local f, g
g = function(...) end
function f()
    local y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")][x["fog"] or "clear"]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]()
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
    y = x[g(if _G.__DEV__ then "blue" else "black")]
    y = if _G.__PROFILING__ then 31337 else 90210
end