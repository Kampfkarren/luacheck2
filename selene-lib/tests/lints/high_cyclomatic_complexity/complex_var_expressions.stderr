error[limit_function_complexity]: cyclomatic complexity is too high (42 > 40)
  ┌─ selene-lib/tests/lints/high_cyclomatic_complexity/complex_var_expressions.lua:4:1
  │
4 │ function f()
  │ ^^^^^^^^^^^^